low-level/framework/platforms/realstor/realstor_enclosure.py

```diff
@@ -223,28 +227,45 @@ def ws_request(self, url, method, retry_count=MAX_RETRIES,
             if response is None:
                 continue
 
-            if response.status_code != self.ws.HTTP_OK:
-
-                # if call fails with invalid session key request or http 403
-                # forbidden request, login & retry
-                if response.status_code == self.ws.HTTP_FORBIDDEN and relogin is False:
-                    logger.info("%s failed, retrying after login " % (url))
-
-                    self.login()
-                    relogin = True
-                    continue
-
-                elif (response.status_code == self.ws.HTTP_TIMEOUT or \
-                         response.status_code == self.ws.HTTP_CONN_REFUSED or \
-                         response.status_code == self.ws.HTTP_NO_ROUTE_TO_HOST) \
-                         and tried_alt_ip is False:
-                    self.switch_to_alt_mc()
-                    tried_alt_ip = True
-                    self.mc_timeout_counter += 1
-                    continue
-            else:
+            if response.status_code == self.ws.HTTP_OK:
+
                 self.mc_timeout_counter = 0
 
+                try:
+                    jresponse = json.loads(response.content)
```

Comment:
For 99% of the api successful response (http 200 & return-code = 0), providing required valid json data, we will still end up doing this check(line 235-239), which is a overhead currently. I dont know how to avoid this, please think & suggest if anyone has any idea.
@ArchanaLimaye you may add a TODO here
> TODO: Need a way to check return-code 2 in more optimal way if possible, currently being checked for all http 200 responses.

Reply:
Done